### DISCUSSION TITLE
Potential use cases for general purpose C wrapper

### DISCUSSION THREAD

**Discussion by jothepro** [cross-language-cpp/djinni-generator#89] [Ideas]
I'd like to collect ideas what would be potential use cases for a general purpose C wrapper. Setting the implementation details aside, what could the generated C interface be used for?
I am asking this because I am currently missing a clear picture on what our long-term goal regarding this topic is.

**Comment by jothepro** [comment]
First thing that comes to my mind: I'm experimenting with developing native UI with GTK on Linux. It can be done with C++ using gtkmm, but it seems to be a lot more fun with [Vala](https://wiki.gnome.org/Projects/Vala/GTKSample). 
Vala is a C preprocessor, so it should be relatively easy to generate Vala gluecode around the C wrapper.

But that would be an entire new language target that would need to be developed and maintained...

**Comment by a4z** [comment]
I use a little bit of go, for some fun ;-)

**Comment by canatella** [comment]
They could help with adding Rust support too: https://github.com/cross-language-cpp/djinni-generator/discussions/75

**Comment by gibutm** [comment]
A use case I'm presently trying is for consumption with [dart:ffi](https://dart.dev/guides/libraries/c-interop) for [Flutter](https://docs.flutter.dev/development/platform-integration/android/c-interop)
